Web.md

```diff
@@ -202,6 +202,92 @@ Given these values, to process a potential WebAssembly response:
+[serializable object](https://html.spec.whatwg.org/multipage/infrastructure.html#serializable-objects), and will require changes to the
+[HTML Standard](https://html.spec.whatwg.org/multipage/infrastructure.html#serialization-steps) as follows:
+
+[StructuredSerializeInternal](https://html.spec.whatwg.org/multipage/infrastructure.html#structuredserializeinternal) adds a new step after step 12:
+
+* Otherwise, if *value* has a [[WebAssembly.Memory]] internal slot, then:
+  1. Let *size* be *value*.[[BufferObject]].[[ArrayBufferByteLength]].
+  2. Let *maximum* be the value of *value*.[[WebAssembly.Memory]]'s `max` field (which is an optional u32).
+  3. If *value*.[[BufferObject]] is a Shared Data Block, then:
+     1. If *forStorage* is true, then throw a
+        "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+        [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
+     2. Otherwise,
+        set *serialized* to { [[Type]]: "WebAssembly.Memory",
+        [[ArrayBufferData]]: *value*.[[BufferObject]].[[ArrayBufferData]],
+        [[ArrayBufferByteLength]]: *size*,
+        [[MemoryMaximum]]: *maximum*,
+        [[AgentCluster]]: the [current Realm Record](https://tc39.github.io/ecma262/#current-realm)'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) }.
+  4. Otherwise throw a
+     "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+     [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
+
+[StructuredDeserialize](https://html.spec.whatwg.org/multipage/infrastructure.html#structureddeserialize) adds a new step after step 12:
+
+* Otherwise, if *serialized*.[[Type]] is "WebAssembly.Memory", then:
+  1. Assert [IsSharedArrayBuffer](https://tc39.github.io/ecma262/#sec-issharedarraybuffer)(*serialized*.[[ArrayBufferData]]) is true.
+  2. If *targetRealm*'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+     *serialized*.[[AgentCluster]], then throw a
+     "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+     [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
+  3. Set *value* to a new WebAssembly.Memory object in *targetRealm*, where:
+     1. *value*'s internal slot [[WebAssembly.Memory]] is set to a new `Memory.memory` *mem*, where:
+        1. *mem*'s linear memory is set to *serialized*.[[BufferObject]].[[ArrayBufferData]].
+        2. *mem*'s linear memory has a byte length of *serialized*.[[BufferObject]].[[ArrayBufferByteLength]].
+        3. *mem*'s `max` size is set tof *serialized*.[[MemoryMaximum]].
+     2. *value*'s internal slot [[BufferObject]] is set to a new `SharedArrayBuffer` *buf*, where:
+        1. *buf*'s internal slot [[ArrayBufferData]]
+           aliases *mem*'s linear memory.
+        2. *buf*'s internal slot [[ArrayBufferByteLength]]
+           is set to the byte length of *mem*'s linear memory.
```

Comment:
Thinking about this, it probably would make sense if the core spec provided an *allochostmem* function to delegate to for creating that memory instance, just like it does for functions. (And similarly for tables and globals.) I can add that.